File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.js
function isPrime(n) {
    if (Math.sqrt(n) === Math.floor(Math.sqrt(n))) return false;
    for (let i = 2; i <= Math.floor(Math.sqrt(n)); i++) {
        if (n % i === 0) return false;
    }
    return true;
}
function solution(n) {
    var answer = 0;
    for (let i = 2; i <= n; i++) {
        if (isPrime(i)) answer++;
    }
    return answer;
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py
from itertools import permutations
import math

def is_prime(num):
    sqrt_num = math.sqrt(num)
    if sqrt_num.is_integer():
        return False
    return all(num % i != 0 for i in range(2, num // 2 + 1))


def solution(numbers):
    result = set()
    str_numbers = list(numbers)
    for i in range(1, len(str_numbers) + 1):
        for perm in permutations(str_numbers, i):
            number = int(''.join(perm))
            if is_prime(number):
                result.add(number)

    return len(result)